config/src/main/java/com/alibaba/nacos/config/server/service/repository/embedded/BaseDatabaseOperate.java

```diff
@@ -234,7 +239,7 @@ default Boolean update(TransactionTemplate transactionTemplate, JdbcTemplate jdb
             } catch (CannotGetJdbcConnectionException e) {
                 FATAL_LOG.error("[db-error] sql : {}, args : {}, error : {}", errSql[0], args[0], e.toString());
                 throw e;
-            } catch (DataAccessException e) {
+            } catch (DataAccessException | IllegalTransactionStateException e) {
```

Comment:
illegal transaction exception should not be logged as DataAccessException,besides ,it is not a fatal exception  and should not be log into FATAL_LOG

Reply:
Modified according to your suggestion
 @shiyiyue1102 please review again.